### ISSUE TITLE
Segfault within modules

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#585]
v2.55.35
```
$ tpl
?- module(v).
Info: created module 'v'
   true.
?- [user]. v:"V". end_of_file.
   true.
?- listing.
'V'(dummy,dummy).
   true.
?- 'V'(X,X).
Segmentation fault (core dumped)

$ tpl
?- [user]. user:"U". end_of_file.
   true.
?- listing.
'U'(dummy,dummy).
   true.
?- 'U'(X,X).
Segmentation fault (core dumped)
```

**Comment by flexoron** [comment]
v2.55.37
```
$ tpl
?- [user]. user:"U". end_of_file.
   true.
?- listing.
'U'.
   true.
?- [user].
"U".
Error: type error, not callable, user:1
   true.
?- listing.
   true.
?- 'U'.
   throw(error(existence_error(procedure,'U'/0),'U'/0)).
?- [user]. user:"U". end_of_file.
Warning: overwriting 'U'/0 % overwriting 'U'/0 which does not exist(and has not been listed)?
   true.
?- 
```

**Comment by flexoron** [comment]
v2.55.38
```
?- [user]. user:"U". end_of_file.
   true.
?- listing.
'U'(dummy,dummy).
   true.
?- [user].
"U".
Error: type error, not callable, user:1
   true.
?- listing.
   true.
?- 'U'(X,Y).
   throw(error(existence_error(procedure,'U'/2),'U'/2)).
?- [user]. user:"U". end_of_file.
Warning: overwriting 'U'/2 % not listed, not existed but overwritten?
   true.
?- 'U'(X,Y).
   X = dummy, Y = dummy.
?- 'U'(X,X).
Segmentation fault (core dumped)
```

**Comment by infradig** [comment]
I'm aware of all these things. No need to keep reposting. It's an ongoing
thing.

**Comment by flexoron** [comment]
Ok.

**Comment by infradig** [comment]
Fix for issue #587 brought this back...

$ tpl
?- module(v).
Info: created module 'v'
   true.
?- [user]. v:"V". end_of_file.
Error: not callable v:(V)/2
   true.
?- listing.
malloc(): unaligned tcache chunk detected
Aborted (core dumped)

EDIT: But lesser of two evils.
